### PR TITLE
Swift 4

### PR DESCRIPTION
**Pull request by pietbrauer**
## Summary

* Uses `Codable` in all models

## Breaking Changes

* `Label`s `color` property is a `String` instead of a `Color`

### CODE REVIEW

File: OctoKit/Issue.swift [hunk 1]
```diff
@@ -3,75 +3,57 @@ import RequestKit
...
-@objc open class Issue: NSObject {
-    open var id: Int
-    open var url: URL?
-    open var repositoryURL: URL?
-    open var labelsURL: URL?
-    open var commentsURL: URL?
-    open var eventsURL: URL?
-    open var htmlURL: URL?
+@objc open class Issue: NSObject, Codable {
+    @objc open private(set) var id: Int = -1
```
Comment (tiferrei) [inline OctoKit/Issue.swift]: Hey @pietbrauer, I'm working on porting my RequestKit based framework to Swift 4 as well, could you please explain why we need the extra words for Codable? Thanks!

Comment (tiferrei) [reply OctoKit/Issue.swift]: Also, `@objc` seems to be deprecated in Swift 4, is there a new alternative to `@objc`? 
Cheers.

Comment (pietbrauer) [reply OctoKit/Issue.swift]: It is not deprecated, it just changed. Before if you marked a class as Objective-C available all its properties where also Objective-C available. In Swift 4 you have to mark them explicitly as Objective-C available so you can access them from Objective-C code.

Comment (tiferrei) [reply OctoKit/Issue.swift]: Got it, thanks!

**Comment by yhkaplan** [comment]
@pietbrauer Hello, I am working on a Swift 4 project that could benefit from this project, but is there something I can do to help update Octokit to Swift 4?

**Comment by yhkaplan** [comment]
@pietbrauer It looks like you're pretty busy, but please feel free to contact me if you get the chance.

**Comment by pietbrauer** [comment]
Hey @yhkaplan, sorry for the late reply. Indeed very busy a the moment. I don't remember what the state of this PR is. I think it is ready to merge @tiferrei?

**Comment by tiferrei** [comment]
@pietbrauer, green light from me. 👍✅

* The dependencies have been updated
* CI is using the new Xcode image
* Updated to user RequestKit's `load` and Swit 4 `Codable`
* All tests are passing.

**Comment by pietbrauer** [comment]
@tiferrei Thanks!

**Comment by yhkaplan** [comment]
@pietbrauer @tiferrei Thank you both!

**Pull request merged**